docs/events_and_streams.md

```diff
@@ -0,0 +1,44 @@
+# The description of Event and Stream definitions
+
+`pg_eventstore` provides classes to prepare the events to be inserted into the eventstore. The most important are:
+
+- `PgEventstore::Event` class which represents an event object
+- `PgEventstore::Stream` class which represents a stream object
+
+## Event object and its defaults
+
+`PgEventstore::Event` has the following attributes:
+
+- `id` - String(UUIDv4, optional, not `nil`). If no provided - the value will be autogenerated.
+- `type` - String(optional, not `nil`). Default is an event's class name. Types which start from `$` indicate system events. It is not recommended to prefix your events types with `$` sign.
+- `global_position` - Integer(optional, read only). Event's global position in the eventstore (the "all" stream position in the popular EventstoreDB). Manually assigning this attribute has no effect. It is internally set when reading events from the database.
+- `stream` - PgEventstore::Stream(optional, read only). A Stream an event belongs to, see description below. Manually assigning this attribute has no effect. It is internally set when appending an event to the given stream or when reading events from the database.
+- `stream_revision` - Integer(optional, read only). A revision of an event inside its stream.
+- `data` - Hash(optional). Event's payload data. For example, if you have a `DescriptionChanged` event class, then you may want to have a description value in the event payload data. Example: `DescriptionChanged.new(data: { 'description' => 'Description of something', 'post_id' => SecureRandom.uuid })`
+- `metadata` - Hash(optional). Event metadata. Event meta information which is not part of an events data payload. Example: `{ published_by: publishing_user.id }`
+- `link_id` - String(UUIDv4, optional, read only). If an event is a link event (link events are pointers to other events), this attribute contains the `id` of the original event. Manually assigning this attribute has no effect. It is internally set when appending an event to the given stream or when reading events from the database.
+- `created_at` - Time(optional, read only). Database's timestamp when an event was appended to a stream. You may want to put your own timestamp into a `metadata` attribute - it may be useful when migrating between different databases. Manually assigning this attribute will affect on nothing. It is internally set when appending an event to the given stream or when reading events from the database.
```

Comment:
```suggestion
- `created_at` - Time(optional, read only). Database's timestamp when an event was appended to a stream. You may want to put your own timestamp into a `metadata` attribute - it may be useful when migrating between different databases. Manually assigning this attribute has no effect. It is internally set when appending an event to the given stream or when reading events from the database.
```